### PR TITLE
Dressing room + grab fixes

### PR DESCRIPTION
**Pull request by sethalves**
- early (incomplete) implementation of a dressing room with a doppelganger dress dummy.
- handControllerGrab bug fixes

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: b1267f9d378e6943bad6437bb8167eaf777db29b available at: https://gist.github.com/44464b1c9c4eb23601f9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 07c70a1ba20410596207bc512353c2732bd92aba available at: https://gist.github.com/44464b1c9c4eb23601f9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d56702028350542cee23b734c369e84a4f9dc79d available at: https://gist.github.com/44464b1c9c4eb23601f9<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7037.dmg

**Comment by nbq** [comment]
Working to test this one now, may ask for testing instructions but I am going to try to wing it.

**Comment by nbq** [comment]
@sethalves I need some direction on how to test this please. I run the new attachedEntitiesManager.js and the new handControllerGrab.js, next I run this script (run through rawgit.com to make sure everything resolves correctly): https://rawgit.com/sethalves/hifi/dressing-room/examples/dressing_room/setupDressingRoom.js and I even wear the Albert body to match the doppleganger (it does not work with other bodies as you can expect).

With all of that said the doppleganger mimics me fine but I cannot drag or do anything to the hat. If reading the comment about it being incomplete is still true then I agree and it passes since the doppleganger works in mimicking your actions.

**Comment by sethalves** [comment]
testing -- don't worry overly about the dressing-room stuff.  try it if you want, but it isn't ready for prime-time.  I needed to get these fixes ready quickly and this branch had the newest version of the grab script.

for grabbing, the fixes are for:

(1) avatar-A equips something, avatar-B does a near grab of the thing and walks away.  The equip should automatically cancel.  When avatar-B releases the entity, it should fall to the ground.

(2) distance-grab something, pull it to your avatar and then near-grab it with the other hand.  stop the distance grab.  throw the entity.  before this PR it would stick in the air.  After, it should be thrown in a ballistic arc, as expected.

(3) create a non-physical entity (such as a web-view).  edit userData and make it grabbable:
{"grabbableKey":{"grabbable":true}}
equip it with one hand, and adjust it with the other.  Before this PR, it would end up stuck to the wrong hand.

File: examples/dressing_room/setupDressingRoom.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+var createPlatformWithLights = Script.resolvePath('createPlatformWithLights.js?'+Math.random(0,100));
```
Comment (ZappoMan) [inline examples/dressing_room/setupDressingRoom.js]: this is not a good design... I'll merge this as is... but it's messed up that you're randomizing here. If we're not reloading the cache when the file changes, it means something else bad is happening!!!

**Pull request merged**